components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AMDefaultKeyManagerImpl.java

```diff
@@ -378,6 +381,21 @@ public AccessTokenInfo getNewApplicationAccessToken(AccessTokenRequest tokenRequ
         return tokenInfo;
     }
 
+    @Override
+    public String getNewApplicationConsumerSecret(AccessTokenRequest tokenRequest) throws APIManagementException {
+        OAuthAdminService oauthAdminService = new OAuthAdminService();
+        OAuthConsumerAppDTO appDTO;
+        try {
+            if (oauthAdminService != null) {
```

Comment:
can oauthAdminService be null?

Reply:
According to the previous discussion, I have added this safe check.

Reply:
but new instance is created just before and it cant be null unless exception is throwed. But in that case flow wont reach this line

Reply:
Yes, we can remove this null check.